transaction/std/src/memo_builder/gift_code_funding_memo_builder.rs

```diff
@@ -48,16 +51,21 @@ impl GiftCodeFundingMemoBuilder {
             )));
         }
         Ok(Self {
-            note: note.into(),
             gift_code_tx_out_public_key: None,
+            fee: Amount::new(Mob::MINIMUM_FEE, Mob::ID),
+            note: note.into(),
             wrote_change_memo: false,
         })
     }
 }
 
 impl MemoBuilder for GiftCodeFundingMemoBuilder {
     /// Set the fee
-    fn set_fee(&mut self, _fee: Amount) -> Result<(), NewMemoError> {
+    fn set_fee(&mut self, fee: Amount) -> Result<(), NewMemoError> {
+        if self.wrote_change_memo {
+            return Err(NewMemoError::FeeAfterChange);
+        }
+        self.fee = fee;
```

Comment:
If any corrective action is done here it may affect my comment in the cancellation memo of let `make_memo_for_change_output` propagate the error.
In the `new()` method above the note was verified for size and an error occurred immediately, however here the `fee` isn't checked and instead an error is propagated later, this lacks symmetry for the users.  Should always check at the builder interface or should always delegate to the memo type